### PR TITLE
[ADDON-43435] don't remove host if other record uses it

### PR DESCRIPTION
**Pull request by okuzhel**
# Description

If 2 different records used same IP and one of them was removed. IP was removed from the mongo

Fixes # [ADDON-43435](https://jira.splunk.com/browse/ADDON-43435)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix

## How Has This Been Tested?

Added 2 entries with same IP but different profile and removed one of them. Log about removing host from mongo db did not appear.

## Checklist

- [x My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] I have run pre-commit on all files before creating the PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: splunk_connect_for_snmp_poller/manager/poller.py [hunk 1]
```diff
@@ -127,22 +130,28 @@ def __check_inventory(self):
...
+    def clean_job_inventory(self, inventory_entry_keys: set, inventory_hosts: set):
         for entry_key in list(self._jobs_map):
-            if entry_key not in inventory_hosts:
+            if entry_key not in inventory_entry_keys:
                 logger.debug("Removing job for %s", entry_key)
                 schedule.cancel_job(self._jobs_map.get(entry_key))
                 db_host_id = return_database_id(entry_key)
-                logger.debug("Removing _id %s from mongo database", db_host_id)
-                self._mongo_walked_hosts_coll.delete_host(db_host_id)
+                host_without_ip = db_host_id.split(":")[0]
```
Comment (omrozowicz-splunk) [inline splunk_connect_for_snmp_poller/manager/poller.py]: shouldn't it be `host_without_port`?

**Review by omrozowicz-splunk — APPROVED** [review]

File: splunk_connect_for_snmp_poller/manager/poller.py (hunk 1 above)
Comment (okuzhel) [reply splunk_connect_for_snmp_poller/manager/poller.py]: yep 😆

Comment (weliasz) [reply splunk_connect_for_snmp_poller/manager/poller.py]: We allow presence of multiple agents per host on different ports. So host:port is unique identifier of an agent.
When adding existing agents inventory_hosts.add(ir.host) uses host from inventory (which could be host:port)
and when checking if mongo entry should be deleted you used if host_without_ip not in inventory_hosts:
What is the reason behind this discrepancy.

Comment (okuzhel) [reply splunk_connect_for_snmp_poller/manager/poller.py]: @weliasz good catch, never used port in inventory so didn;t think about that use case, changed

**Pull request merged**